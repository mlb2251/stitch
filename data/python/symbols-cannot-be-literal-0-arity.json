[
  "(+ 1 2 3 &a &b &c 4 5 6 7 8 9 10)",
  "(+ 1 2 3 &a &b &c 4 5 6 7 8 9 10)"
]