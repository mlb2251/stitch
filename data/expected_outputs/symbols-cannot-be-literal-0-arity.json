{
  "cmd": "/home/kavi/mit/stitch/target/debug/deps/integration_tests-e3a9d4911874cf31 symbols_basic",
  "args": {
    "iterations": 2,
    "abstraction_prefix": "fn_",
    "previous_abstractions": 0,
    "shuffle": false,
    "truncate": null,
    "no_opt": false,
    "silent": false,
    "verbose_rewrite": false,
    "step": {
      "max_arity": 3,
      "threads": 1,
      "no_stats": false,
      "batch": 1,
      "dynamic_batch": false,
      "eta_long": false,
      "no_curried_metavars": false,
      "no_curried_bodies": false,
      "inv_candidates": 1,
      "hole_choice": "DepthFirst",
      "cost": {
        "cost_lam": 1,
        "cost_app": 1,
        "cost_var": 100,
        "cost_ivar": 100,
        "cost_prim_default": 100,
        "cost_prim": "{}"
      },
      "no_mismatch_check": false,
      "follow": null,
      "follow_prune": false,
      "verbose_worklist": false,
      "verbose_best": false,
      "print_stats": 0,
      "show_rewritten": false,
      "rewritten_dreamcoder": false,
      "rewritten_intermediates": false,
      "inv_arg_cap": false,
      "allow_single_task": false,
      "no_opt_single_use": false,
      "no_opt_upper_bound": false,
      "no_opt_force_multiuse": false,
      "no_opt_useless_abstract": false,
      "no_opt_arity_zero": false,
      "no_other_util": false,
      "structure_penalty": 1.0,
      "rewrite_check": false,
      "utility_by_rewrite": false,
      "dreamcoder_comparison": false,
      "quiet": false,
      "fused_lambda_tags": {
        "tags": null
      },
      "tdfa": {
        "tdfa_json_path": "",
        "tdfa_root": "",
        "valid_metavars": "",
        "valid_roots": "",
        "tdfa_non_eta_long_states": ""
      }
    }
  },
  "original_cost": 2826,
  "final_cost": 2220,
  "compression_ratio": 1.272972972972973,
  "num_abstractions": 1,
  "original": [
    "(+ 1 2 3 &a &b &c 4 5 6 7 8 9 10)",
    "(+ 1 2 3 &a &b &c 4 5 6 7 8 9 10)"
  ],
  "rewritten": [
    "(fn_0 &a &b &c 4 5 6 7 8 9 10)",
    "(fn_0 &a &b &c 4 5 6 7 8 9 10)"
  ],
  "rewritten_dreamcoder": null,
  "abstractions": [
    {
      "body": "(+ 1 2 3)",
      "dreamcoder": "#(+ 1 2 3)",
      "arity": 0,
      "name": "fn_0",
      "utility": 203,
      "final_cost": 2220,
      "compression_ratio": 1.272972972972973,
      "cumulative_compression_ratio": 1.272972972972973,
      "num_uses": 2,
      "rewritten": null,
      "rewritten_dreamcoder": null,
      "uses": [
        {
          "fn_0 ": "(+ 1 2 3)"
        }
      ],
      "dc_comparison_millis": null
    }
  ]
}